python/ml_wrappers/model/predictions_wrapper.py

```diff
@@ -0,0 +1,199 @@
+
+        :param test_data: The data that was used to train the model.
+        :type test_data: pd.DataFrame
+        :param y_pred: Predictions of the model.
+        :type y_pred: np.ndarray
+        """
+        if not isinstance(test_data, pd.DataFrame):
+            raise DataValidationException(
+                "Expecting a pandas dataframe for test_data")
+        if not isinstance(y_pred, np.ndarray):
+            raise DataValidationException(
+                "Expecting a numpy array for y_pred")
+        if len(test_data) != len(y_pred):
+            raise DataValidationException(
+                "The number of instances in test data "
+                "do not match with number of predictions")
+        self._combined_data = test_data.copy()
+        self._combined_data[TARGET] = y_pred
+
+    def _get_filtered_data(
+            self, query_test_data_row: pd.Series) -> pd.DataFrame:
+        """Return the filtered data based on the query data.
+
+        :param query_test_data_row: The query data instance.
+        :type query_test_data_row: pd.Series
+        :return: The filtered dataframe based on the values in query data.
+        :rtype: pd.DataFrame
+        """
+        queries = []
+        for column_name, column_data in query_test_data_row.iteritems():
+            if isinstance(column_data, str):
+                queries.append("`{}` == '{}'".format(
+                    column_name, column_data))
+            else:
+                queries.append("`{}` == {}".format(
+                    column_name, column_data))
+
+        queries_str = '(' + ') & ('.join(queries) + ')'
+        filtered_df = self._combined_data.query(queries_str)
+
+        if len(filtered_df) == 0:
+            raise EmptyDataException(
+                "The query data was not found in the combined dataset")
+
+        return filtered_df
+
+    def predict(self, query_test_data: pd.DataFrame) -> np.ndarray:
+        """Return the predictions based on the query data.
+
+        :param query_test_data: The data for which the predictions need to
+            be returned.
+        :type query_test_data: pd.DataFrame
+        :return: Predictions of the model.
+        :rtype: np.ndarray
+        """
+        if not isinstance(query_test_data, pd.DataFrame):
+            raise DataValidationException(
+                "Expecting a pandas dataframe for query_test_data")
+        prediction_output = []
+        for _, row in query_test_data.iterrows():
+            filtered_df = self._get_filtered_data(row)
+            prediction_output.append(filtered_df[TARGET].values[0])
+
+        return np.array(prediction_output)
+
+    def __setstate__(self, state):
+        """Set the state so that the wrapped model is
+        serializable via pickle."""
+        self._combined_data = state["_combined_data"]
+
+    def __getstate__(self):
+        """Return the state so that the wrapped model is
+        serializable via pickle."""
```

Comment:
do you want to add the return value/type in the doc here

Reply:
Done